src/Tutor.Web/Controllers/Analytics/AnalyticsController.cs

```diff
@@ -104,7 +109,7 @@ public ActionResult<PagedResult<DomainEvent>> GetEvents([FromQuery] int page, [F
     [HttpGet("all-events")]
     public ActionResult<List<DomainEvent>> GetAllEvents()
     {
-        var result = _eventStore.GetAllEvents();
+        var result = _eventStore.Events.ToList();
```

Comment:
The changes made here seem to have significantly increased the performance. GetAllEvents completes in about a second, while KcStatistics are compiled in less than a second.

I've changed the ImmutableDictionary to a regular dictionary to see if this affects performance in any way and this was not the case.

For GetAllEvents, the change made here essentially reorganizes identical logic, so I am not sure what is responsible for the performance gain.

As a side note, the current export has an issue when compared to the old one (see images below).
**Current**
![image](https://user-images.githubusercontent.com/7092212/188072986-c549f9f9-a2ec-4991-9a68-d8db4e9f2fd6.png)
**Older**
![image](https://user-images.githubusercontent.com/7092212/188073022-9558af61-4ff7-415a-b880-8d9aa3ec5a59.png)

Reply:
I think I figured out what was causing the performance issues, and it's honestly a pretty stupid mistake - the old version of EventSerializer used to get the serialization options as EventSerializationConfiguration.GetEventSerializerOptions() for every single call to Serialize or Deserialize. This means that the entire JsonSerializerOptions object was instantiated and configured once per event, and I definitely didn't notice just how inefficient that is when I initially wrote the code, but look at it now, just configuring the AllowedTypesDiscriminatorConvention includes copying multiple dictionaries, and I don't even know what exactly the Dahomey.Json initialization methods do and how costly they might be. 

I think the export issue is because of the way the serializer for the controllers is configured, I assume the export expects '$discriminator' instead of 'typeDiscriminator'?

Reply:
@lukaDoric can you comment on the export?

Reply:
I changed it back to '$discriminator' in 02acfa3104adbf25dc1aaf41773e57833916d87c, maybe retry the export now? I would try it but I don't know where the export functionality is or how to use it.